src/Avalonia.Controls/TopLevel.cs

```diff
@@ -144,10 +146,11 @@ public TopLevel(ITopLevelImpl impl)
         /// </param>
         public TopLevel(ITopLevelImpl impl, IAvaloniaDependencyResolver? dependencyResolver)
         {
+            _appRoot ??= new WeakReference<TopLevel>(this);
```

Comment:
I don't think it's a good approach. Something as simple as a splash screen that lives for 1 second will break it.

Reply:
Good catch, I hadn't thought of that possibility. The problem is that without IApplicationLifieTime, it is not possible to correlate the App to the TopLevels. Do you have any suggestions?